server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -225,21 +236,33 @@ public static class FileSettingsHealthIndicatorService implements HealthIndicato
             )
         );
 
-        private final AtomicLong changeCount = new AtomicLong(0);
-        private final AtomicLong failureStreak = new AtomicLong(0);
-        private final AtomicReference<String> mostRecentFailure = new AtomicReference<>();
+        private boolean isActive = false;
+        private long changeCount = 0;
+        private long failureStreak = 0;
+        private String mostRecentFailure = null;
 
-        public void changeOccurred() {
-            changeCount.incrementAndGet();
+        public synchronized void startOccurred() {
```

Comment:
I got tired of worrying about mixing and matching the states of these fields when they're changed independently. Now they're all just protected by `synchronized`.